src/core.js

```diff
@@ -272,6 +272,16 @@ $.extend( $.validator, {
 				this.element( element );
 			}
 		},
+		oninput: function( element ) {
+			if ( element.name in this.submitted || element.name in this.invalid ) {
```

Comment:
`input` event fires only when the value changes. The empty val check in `onkeyup` was added to prevent validating empty element when losing focus. So I don't see a need for it here.

But I'm open to suggestion.